lib/awful/placement.lua

```diff
@@ -800,8 +801,18 @@ end
 -- Check if the proposed geometry fits the screen
 local function fit_in_bounding(obj, geo, args)
     local sgeo   = get_parent_geometry(obj, args)
+    for key, value in pairs(sgeo) do
+        if type(value) == "number" then
+            sgeo[key] = gmath.round(value)
+        end
+    end
```

Comment:
The data structure for `cairo.RectangleInt` parameter is `{ x, y, height, width }`. I think we better have a utility function `round_geometries(geo) -> rounded_geo`. (It would also prevent code duplication for the similar loops you added)

Reply:
Thanks for the review!
```lua
local function round_geometry(geo)
    local geo_keys = { "x", "y", "height", "width" }
    local rounded_geo = {}
    for _, key in pairs(geo_keys) do
        rounded_geo[key] = gmath.round(geo[key])
    end
    return rounded_geo
end

-- Check if the proposed geometry fits the screen
local function fit_in_bounding(obj, geo, args)
    local round_sgeo   = round_geometry(get_parent_geometry(obj, args))
    local region = cairo.Region.create_rectangle(cairo.RectangleInt(round_sgeo))

    local round_geo = round_geometry(geo)
    region:intersect(cairo.Region.create_rectangle(
        cairo.RectangleInt(round_geo)
    ))

    local geo2 = region:get_rectangle(0)

    -- If the geometry is the same then it fits, otherwise it will be cropped.
    return geo2.width == round_geo.width and geo2.height == round_geo.height
end
```
Suppose I write as above.
For `return`, is it better to use `round_geo` or `geo`?

Reply:
I think you should use `rounded_geo` since `geo2` is obtained from `region` that was generated from `round_sgeo`.

Side note : I think we should have a test on `geo[key]` before the assignation in round_geometry.

Reply:
> For `return`, is it better to use `round_geo` or `geo`?

I checked and found that the value of `geo2` is an integer, so I used `round_geo`.
It worked well in local. I force push it.

Reply:
I missed the reply. I will add a test.

Reply:
```lua
    for _, key in pairs(geo_keys) do
        assert(type(geo[key]) == "number")
        rounded_geo[key] = gmath.round(geo[key])
    end
```
I put in an assertion because I think the default value creates confusion.

Reply:
Hm... I was thinking more of something like `if geo[key] ~= nil then` to allow the function to work with "partial geometries"

Reply:
Oh, I see. I'll do it. Thank you.